src/reader/mod.rs

```diff
@@ -1930,3 +1933,68 @@ mod test {
     pub(super) use check;
     pub(super) use small_buffers;
 }
+
+fn read_to_string_impl<R: BufRead>(starting_buffer_position: usize, reader: &mut R, parser: Parser, end: QName, buf: &mut Vec<u8>) -> Result<Vec<u8>> {
```

Comment:
What if anything is `buf` actually used for?  It seems that we copy the data but throw it away in favor of a buffer managed separately and internally, and without any way to reduce the size as the user-provided buffer allows.  

I'm not completely against the idea as I've had a similar one in the past, but I don't want to bolt on such a significant architectural element for this one particular feature without taking other aspects into consideration and thinking about whether it impacts the wisdom of our other API choices.

It also feels a bit like we're reimplementing https://gitlab.com/sequoia-pgp/sequoia/-/tree/main/buffered-reader

Reply:
Yes that's right. `buf` is entirely just to store the temporary contents and allow us to use `reader.read_event_into`. However, the same thing could be said about the current function `reader.read_to_end_into`. The `buf` parameter passed into that function is also just used for temporary holding.

The sequoia BufferedReader is similar to what we have. But the `StoredReader` that we have is much simpler as it just needs to store the read output into a user-provided buffer.